app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+# post
+class Post < ApplicationRecord
+  after_create :posts_counter_update
```

Comment:
- [ ] [OPTIONAL] Good job using Rails Active Callbacks. However, using `after_create` can cause a problem. `after_create` calls the `posts_counter_update` method whenever a post is created but not saved in the database. 

The problem can be caused if an error is thrown from the database itself when creating the record (could be a connection error or any other reason), so the newly created post will be counted for the user but wil not be saved in the database (there will be no record of it).

You can read more about this [here](https://stackoverflow.com/questions/33890458/difference-between-after-create-after-save-and-after-commit-in-rails-callbacks).

Reply:
thanks a lot @ShadyShawkat, I hope I'll find an alternative. 🙌